controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -389,6 +389,30 @@ class ControllerManager : public rclcpp::Node
     const std::vector<ControllerSpec> & controllers, int strictness,
     const ControllersListIterator controller_it);
 
+  /// A functor to be used in the std::sort method to sort the controllers to be able to
+  /// execute them in a proper order
+  /**
+   * @brief controller_sorting - A functor that compares the controllers ctrl_a and ctrl_b and then
+   * returns which comes first in the sequence
+   *
+   *  @note The following conditions needs to be handled while ordering the controller list
+   *  1. The controllers that do not use any state or command interfaces are updated first
+   *  2. The controllers that use only the state system interfaces only are updated next
+   *  3. The controllers that use any of an another controller's reference interface are updated
+   * before the preceding controller
+   *  4. The controllers that use the controller's estimated interfaces are updated after the
+   * preceding controller
+   *  5. The controllers that only use the system's command interfaces are updated last
+   *  6. All inactive controllers go at the end of the list
```

Comment:
> Should we go with the “new” nomenclature here?

Maybe it would be easier to have the renaming to the new nomenclature everything at once. As it is not started, it might create confusion to the community.

> Should we first merge “estimated” interfaces and then do rename, or do the renaming first.

I think it is better to merge the "estimated" interfaces first and then do the renaming together (or) may be we can start a renaming branch on top of the estimated interfaces branch, that would also work right? 

> What do you mean with “system's” command interfaces? Do you mean HW command interfaces?

Yes, you are right. I mean HW command interfaces. I will be changing it.

> This docs should be probably moved somewhere else.

Where do you think would be the right place.